### PR TITLE
Added  go report badge

### LINKED ISSUE #3209: Add Go report and build passing badges in CAPA project
/kind feature

**Describe the solution you'd like**

We can add `build passing` and `go report` badge in CAPA project.

This will add following badges in CAPA README.md
</a>
Build Status</a>

### PR DESCRIPTION
**Pull request by shivi28**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR adds `go report` badges in CAPA project.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3209 

**Special notes for your reviewer**:
Now we can see following badges in CAPA repo.
</a>

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Added go report badge in CAPA
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,15 @@
 <img src="https://github.com/kubernetes/kubernetes/raw/master/logo/logo.png"  width="100x"><a href="https://aws.amazon.com/opensource/"><img width="192x" src="https://d0.awsstatic.com/logos/powered-by-aws.png" alt="Powered by AWS Cloud Computing"></a>
 </p>
 <p align="center">
+<!-- build passing badge -->
+<a href="https://prow.k8s.io/?job=post-cluster-api-provider-aws-push-images">
+<img alt="Build Status" src="https://prow.k8s.io/badge.svg?jobs=post-cluster-api-provider-aws-push-images"></a>
```
Comment (sedefsavas) [inline README.md:9]: `https://prow.k8s.io/?job=post-cluster-api-provider-aws-push-image` is not showing any jobs.
Also it is not the build job.

**Review by sedefsavas — COMMENTED** [review]
Also, adding go reference badge is useful like CAPI has.
https://pkg.go.dev/sigs.k8s.io/cluster-api-provider-aws

**Comment by shivi28** [comment]
> Also, adding go reference badge is useful like CAPI has. https://pkg.go.dev/sigs.k8s.io/cluster-api-provider-aws

Yes, Go reference we already have in CAPA's Readme.md

File: README.md (hunk 1 above)
Comment (shivi28) [reply README.md:9]: Today we are not seeing any build list for any provider on prow jobs. I checked in CAPI repository as well. CAPI's build list is also empty `https://prow.k8s.io/?job=post-cluster-api-push-images`.

I will post my findings once I will get some more hints on why it's happening

Comment (shivi28) [reply README.md]: I have removed `build passing` badge  as the core of our tests are in prow and we might get same `build | no results` multiple times, so feels it's not good to show it on repo

**Pull request merged**